### PR TITLE
Allowing to set several namespaces during one session

### PR DESCRIPTION
**Pull request by vincentdieltiens**
This PR gives the possibility to connect to one or more namespaces during a connection.

```php
$client = new Client(
	new Version1X($url)
);
$client->initialize();
// Send message to /namespace1
$client->of('/namespace1');
$client->emit('message', [ 'message' => 'Hello world']);

// Send message to /namespace2
$client->of('/namespace2');
$client->emit('message', [ 'message' => 'Hello world']);

// Reset to no namespace
$client->of('');
$client->emit('message', [ 'message' => 'Hello world']);
$client->close();
```

### CODE REVIEW

File: src/EngineInterface.php [hunk 1]
```diff
@@ -57,5 +57,8 @@ public function keepAlive();
 
     /** Gets the name of the engine */
     public function getName();
+
+    /** Sets the namespace for the next messages */
```
Comment (Taluu) [inline src/EngineInterface.php]: missing doc for the `$namespace` parameter (even though we can guess it really easily)

**Comment by Taluu** [comment]
This looks good ! Could you add an example in the `example` directory ? Thanks !

**Comment by vincentdieltiens** [comment]
I just created the new examples but it seems that the path to the vendor autoload for the other examples are incorrect ?

**Comment by Taluu** [comment]
Assuming that you run the examples only from the elephant.io root, I assumed that you ran composer from the root of Elephant.io, not the root of the project using Elephant..

**Comment by vincentdieltiens** [comment]
Actually, I ran composer outside Elephant.io (it is a dependency of a testing project). I just run composer inside Elephant.io project, it works ;-)

**Comment by vincentdieltiens** [comment]
Sorry :D

**Comment by vincentdieltiens** [comment]
Anything else to change ?

**Comment by Taluu** [comment]
Yep, I'll do a manual merge when I will have some time (probably during the week), as I need to make tiny changes (such as some squashes, ...)

**Comment by Taluu** [comment]
Sorry for the delay, it's been manually merged into master. :}

**Pull request closed without merging**